src/webapp/components/survey-list/SurveyListTable.tsx

```diff
@@ -0,0 +1,574 @@
+import { Survey, SurveyBase, SURVEY_FORM_TYPES } from "../../../domain/entities/Survey";
+import styled from "styled-components";
+import {
+    TableBody,
+    TableContainer,
+    Table,
+    TableRow,
+    Paper,
+    TableCell,
+    TableHead,
+    Typography,
+} from "@material-ui/core";
+import i18n from "@eyeseetea/feedback-component/locales";
+import { ActionMenuButton } from "../action-menu-button/ActionMenuButton";
+import { palette } from "../../pages/app/themes/dhis2.theme";
+import { Id } from "../../../domain/entities/Ref";
+import { getChildSurveyType, getSurveyOptions } from "../../../domain/utils/PPSProgramsHelper";
+import { useHistory } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { ArrowDownward, ArrowUpward } from "@material-ui/icons";
+import _ from "../../../domain/entities/generic/Collection";
+
+interface SurveyListTableProps {
+    surveys: Survey[] | undefined;
+    surveyFormType: SURVEY_FORM_TYPES;
+    updateSelectedSurveyDetails: (
+        survey: SurveyBase,
+        orgUnitId: Id,
+        rootSurvey: SurveyBase
+    ) => void;
+}
+
+export type SortDirection = "asc" | "desc";
+export type SurveyColumns = keyof Survey;
+export const SurveyListTable: React.FC<SurveyListTableProps> = ({
+    surveys,
+    surveyFormType,
+    updateSelectedSurveyDetails,
+}) => {
+    const [options, setOptions] = useState<string[]>([]);
```

Comment:
Try to separate presentation logic from render logic creating a custom hook. Components just show info and a custom hook decides.

Reply:
done in https://github.com/EyeSeeTea/amr-surveys/pull/11